wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html

```diff
@@ -14,7 +14,7 @@
     >
         <div class="w-dialog__overlay" data-action="click->w-dialog#hide"></div>
         <div class="w-dialog__box">
-            <button type="button" class="w-dialog__close-button" aria-label="{% trans 'Close dialog' %}" data-action="w-dialog#hide">
+            <button type="button" class="w-dialog__close-button{% if message_heading and message_icon_name %} w-text-grey-600{% endif %}" aria-label="{% trans 'Close dialog' %}" data-action="w-dialog#hide">
```

Comment:
This did not solve the problem exactly as its styling the button and not the icon, however, I have made a refinement as follows;

1. Add a main dialog class for when message is showing
2. When this class exists, override the default colour with this variant of grey